### PR TITLE
Docs: Update section on inspecting tables

### PR DESCRIPTION
**Pull request by wypoon**
Using the SparkSessionCatalog to query metadata tables is possible in Spark 3.2.

### CODE REVIEW

**Comment by wypoon** [comment]
@kbendick @RussellSpitzer @jackye1995 mind reviewing this?

File: docs/versioned/spark/spark-queries.md [hunk 1]
```diff
@@ -168,7 +168,7 @@ To inspect a table's history, snapshots, and other metadata, Iceberg supports me
 Metadata tables are identified by adding the metadata table name after the original table name. For example, history for `db.table` is read using `db.table.history`.
 
 {{< hint info >}}
-As of Spark 3.0, the format of the table name for inspection (`catalog.database.table.metadata`) doesn't work with Spark's default catalog (`spark_catalog`). If you've replaced the default catalog, you may want to use `DataFrameReader` API to inspect the table. 
+In Spark 3.0 and 3.1, if you have replaced Spark's default catalog (`spark_catalog`) with Iceberg's `SparkSessionCatalog`, you cannot use it to query metadata tables, as the form of the metadata table name (`catalog.database.table.metadata`) is not accepted by Spark. This is fixed in Spark 3.2 by [SPARK-34209](https://issues.apache.org/jira/browse/SPARK-34209). For Spark 3.0 and 3.1, you can configure a different catalog (implemented by `SparkCatalog`) to query metadata tables, or you can use the `DataFrameReader` API.
```
Comment (RussellSpitzer) [inline docs/versioned/spark/spark-queries.md:171]: I would just make the wording a bit more specific here. There are a lot of pronouns which I think is a bit difficult to reason about. I also don't think we need the Jira link since that would mostly help dev's and note users, but feel free to keep it if you think it adds value.

In Spark 3.0 and 3.1 the Spark Session catalog (`spark_catalog`) does not support table names with multipart identifiers such as `catalog.database.table.metadata`. To work around the lack of support of multipart identifiers, configure a non-session catalog using the Iceberg SparkCatalog class or use the Spark `DataFrameReader` API. Spark 3.2 and onwards supports multipart identifiers in the Spark Session catalog.

Comment (wypoon) [reply docs/versioned/spark/spark-queries.md:171]: Thanks for the review and suggestion.

**Comment by wypoon** [comment]
I have updated the text. @RussellSpitzer I tweaked your suggested text slightly, and dropped the reference to the SPARK JIRA.

File: docs/versioned/spark/spark-queries.md [hunk 2]
```diff
@@ -168,7 +168,7 @@ To inspect a table's history, snapshots, and other metadata, Iceberg supports me
 Metadata tables are identified by adding the metadata table name after the original table name. For example, history for `db.table` is read using `db.table.history`.
 
 {{< hint info >}}
-As of Spark 3.0, the format of the table name for inspection (`catalog.database.table.metadata`) doesn't work with Spark's default catalog (`spark_catalog`). If you've replaced the default catalog, you may want to use `DataFrameReader` API to inspect the table. 
+In Spark 3.0 and 3.1, the Spark session catalog (`spark_catalog`) does not support table names with multipart identifiers such as `catalog.database.table.metadata`. To work around this, for querying metadata tables, configure a different catalog that uses the Iceberg `SparkCatalog` class, or use the Spark `DataFrameReader` API. From Spark 3.2 onwards, the session catalog supports table names with multipart identifiers.
```
Comment (kbendick) [inline docs/versioned/spark/spark-queries.md:171]: Nit: Consider `Prior to Spark 3.2` instead of `Spark 3.0 and Spark 3.1`.

I don't think this applies at all to Spark 2.4, but I can still imagine getting questions about it.

Comment (wypoon) [reply docs/versioned/spark/spark-queries.md:171]: Spark 2.4 does not support querying tables using SQL, so this does not apply to 2.4 at all. That is why I wrote "In Spark 3.0 and 3.1", since querying using Spark SQL is only supported in Spark 3 onward.
One can argue both ways about "Prior to Spark 3.2": The reader knows that SQL is not supported for Spark 2, so implicitly "prior to Spark 3.2" means 3.0 and 3.1. Or "prior to Spark 3.2" implies that this section applies to 2.4, which would be misleading.

Comment (wypoon) [reply docs/versioned/spark/spark-queries.md]: @kbendick I updated the text.
I added a line about Spark 2.4, pointing the reader to the section on "Inspecting with DataFrames".
I also used "prior to 3.2".

**Comment by wypoon** [comment]
@RussellSpitzer @kbendick mind reviewing this again? I have updated the text and rebased on master (after Sam Redai's reorg of docs). I hope it can be merged now.

**Comment by wypoon** [comment]
@RussellSpitzer @kbendick mind reviewing this again? I have updated the text and rebased on master (after Sam Redai's reorg of docs). I hope it can be merged now.

File: docs/spark/spark-queries.md [hunk 3]
```diff
@@ -168,7 +168,9 @@ To inspect a table's history, snapshots, and other metadata, Iceberg supports me
 Metadata tables are identified by adding the metadata table name after the original table name. For example, history for `db.table` is read using `db.table.history`.
 
 {{< hint info >}}
-As of Spark 3.0, the format of the table name for inspection (`catalog.database.table.metadata`) doesn't work with Spark's default catalog (`spark_catalog`). If you've replaced the default catalog, you may want to use `DataFrameReader` API to inspect the table. 
+For Spark 2.4, use the `DataFrameReader` API to [inspect tables](#inspecting-with-dataframes).
+
+For Spark 3, prior to 3.2, the Spark session catalog (`spark_catalog`) does not support table names with multipart identifiers such as `catalog.database.table.metadata`. To work around this, for querying metadata tables, configure a different catalog that uses the Iceberg `SparkCatalog` class, or use the Spark `DataFrameReader` API. From Spark 3.2 onwards, the session catalog supports table names with multipart identifiers.
```
Comment (szehon-ho) [inline docs/spark/spark-queries.md:173]: Do we need 'spark_catalog' quote here?   It seems a bit confusing, as we are talking about Spark Session Catalog.  It was probably from before, when mentioning the default catalog.  If not needed I feel we can remove it.

Comment (szehon-ho) [reply docs/spark/spark-queries.md:173]: Also I feel like it is still a bit wordy, can we just say

As a workaround, configure the catalog `org.apache.iceberg.spark.SparkCatalog` or use the DataFrameReader API.

It's more consistent with the previous sentence.  I was thinking we dont need 'for querying metadata tables', as I think that's what this section is about.   And I feel, no need to mention From Spark 3.2 onwards, as it's implied the issue is gone after 3.2.

Comment (wypoon) [reply docs/spark/spark-queries.md]: @szehon-ho thanks for the review.
I think that to be correct, one should say "configure a catalog that uses `org.apache.iceberg.spark.SparkCatalog`". The catalog is `prod` or `iceberg_catalog` or whatever one names it. Otherwise, I have adopted your suggestion.

Note: If you don't feel that the link for "session catalog" (in my update) is necessary, I'm fine with removing it.

File: docs/spark/spark-queries.md [hunk 4]
```diff
@@ -168,7 +168,9 @@ To inspect a table's history, snapshots, and other metadata, Iceberg supports me
 Metadata tables are identified by adding the metadata table name after the original table name. For example, history for `db.table` is read using `db.table.history`.
 
 {{< hint info >}}
-As of Spark 3.0, the format of the table name for inspection (`catalog.database.table.metadata`) doesn't work with Spark's default catalog (`spark_catalog`). If you've replaced the default catalog, you may want to use `DataFrameReader` API to inspect the table. 
+For Spark 2.4, use the `DataFrameReader` API to [inspect tables](#inspecting-with-dataframes).
+
+For Spark 3, prior to 3.2, the Spark [session catalog](../spark-configuration#replacing-the-session-catalog) does not support table names with multipart identifiers such as `catalog.database.table.metadata`. As a workaround, configure a catalog that uses `org.apache.iceberg.spark.SparkCatalog`, or use the Spark `DataFrameReader` API.
```
Comment (szehon-ho) [inline docs/spark/spark-queries.md:173]: The second sentence is still a bit long and not too clear (configure a catalog that uses the Iceberg 'SparkCatalog' class) , can we say

```
As a workaround, configure the Iceberg `SparkCatalog` class...
```

I am thinking that especially with the link, it should be enough for them to know how to do it

**Comment by szehon-ho** [comment]
Thanks, the link is fine with me if it works!  I added one more small rewording, then I think the change will be ok

File: docs/spark/spark-queries.md (hunk 4 above)
Comment (wypoon) [reply docs/spark/spark-queries.md:173]: I have to disagree. I think the second sentence is both clear and correct.
Does "configure a catalog using `org.apache.iceberg.spark.SparkCatalog`" sound better to your ears?

Comment (wypoon) [reply docs/spark/spark-queries.md]: Ok, I changed it to simply "configure an `org.apache.iceberg.spark.SparkCatalog`".

About the link, earlier in the same file, there is a link "(../spark-configuration#using-catalogs)"; I followed the same syntax. The links don't work in github, but they should be correct on the Apache Iceberg website.

Comment (szehon-ho) [reply docs/spark/spark-queries.md]: Yea that works (can we change it to 'a' as org.apache... does not start with a vowel?).  I see what you mean, but I guess confusion to me was , 'a catalog using the class org.apache.iceberg.spark.SparkCatalog', seemed to indicate another catalog that uses this internally.

OK I hope so about the link.

**Review by szehon-ho — APPROVED** [review]
Looks good to me, @samredai if you want to have a sanity check?

**Review by samredai — APPROVED** [review]
LGTM! I like the conciseness reached after the reviews, the final wording is great.

**Comment by szehon-ho** [comment]
Merged, thanks @wypoon for the doc, and all the reviewers @kbendick @RussellSpitzer @samredai

**Comment by wypoon** [comment]
Thanks @szehon-ho!
@samredai is it possible to get this update on the website? Iceberg 0.13 supports Spark 3.2, so this information is relevant and helpful to users of the current release.

**Comment by samredai** [comment]
@wypoon if you open this change as a hotfix directly against the 0.13.1 branch in the apache/iceberg-docs repo, that will get auto-deployed once it's merged.
